include/pybind11/stl/filesystem.h

```diff
@@ -13,21 +13,30 @@
 #include <string>
 
 #ifdef __has_include
-#    if defined(PYBIND11_CPP17) && __has_include(<filesystem>)
-#        include <filesystem>
-#        define PYBIND11_HAS_FILESYSTEM 1
+#    if defined(PYBIND11_CPP17)
```

Comment:
This doesn't match experimental optional. Experimental optional can work even if only CPP14 is defined see: https://github.com/pybind/pybind11/blob/9aa676d38db4b18d9952c70486412d20c0213f13/include/pybind11/detail/common.h#L184

Reply:
Actually, nvm as this file has C++17 syntax that would break and complicate the implementation anyhow.